src/Raven.Server/Documents/Handlers/Debugging/NodeDebugHandler.cs

```diff
@@ -35,5 +42,98 @@ public Task ListRemoteConnections()
             }
             return Task.CompletedTask;
         }
+
+        [RavenAction("/admin/debug/node/engine-logs", "GET", AuthorizationStatus.Operator, IsDebugInformationEndpoint = true)]
+        public Task ListRecentEngineLogs()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var write = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(write,ServerStore.Engine.InMemoryDebug.ToJson());
+                write.Flush();
+            }
+            return Task.CompletedTask;
+        }
+        
+        [RavenAction("/admin/debug/node/ping", "GET", AuthorizationStatus.Operator, IsDebugInformationEndpoint = true)]
+        public async Task PingTest()
+        {
+            var dest = GetStringQueryString("url", false) ?? GetStringQueryString("node", false);
+            var topology = ServerStore.GetClusterTopology();
+            var tasks = new List<Task<(string url, string res)>>();
+            if (string.IsNullOrEmpty(dest))
+            {
+                foreach (var node in topology.AllNodes)
+                {
+                    tasks.Add(PingOnce(node.Value));
+                }
+            }
+            else
+            {
+                var url = topology.GetUrlFromTag(dest);
+                tasks.Add(PingOnce(url ?? dest));
+            }
+            
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var write = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                write.WriteStartObject();
+                while (tasks.Count > 0)
+                {
+                    var task = await Task.WhenAny(tasks);
+                    tasks.Remove(task);
+                    var res = task.Result;
+                    write.WritePropertyName(res.url);
+                    write.WriteString(res.res);
+                    if (tasks.Count > 0)
+                    {
+                        write.WriteComma();
+                    }
+                    write.Flush();
+                }
+                write.WriteEndObject();
+                write.Flush();
+            }
+        }
+
+        private async Task<(string url, string res)> PingOnce(string url)
```

Comment:
I don't like this.
A) use PascalCase for tuple args
B) Use proper structure for the reply, not comma separated string, it wouldn't be readable in a year.
C) `Task<(string Url, (int TcpInfoTime, int SendTime, in RecieveTime), string Error)>` would do.